packages/create-sitecore-jss/src/templates/angular-sxp/scripts/config/plugins/disconnected.ts

```diff
@@ -20,7 +20,9 @@ class DisconnectedPlugin implements ConfigPlugin {
       );
     }
 
-    return config;
+    return Object.assign({}, config, {
+      sitecoreApiKey: config.sitecoreApiKey || 'no-api-key-set',
+    });
```

Comment:
@ambrauer This is true, but for me seems like it's confusing that you can start your app in connected mode and apiKey is set by default to 'no-api-key-set', so you will not have any errors (but apiKey is invalid), while if it's empty you will have an error that will be thrown by _clientFactory_.
For disconnected mode it's fine I think, since we need an apiKey and it can be any.
What do you think?

Reply:
@ambrauer I'm talking specifically about this [check](https://github.com/Sitecore/jss/blob/feature/JSS-3546/packages/create-sitecore-jss/src/templates/angular/src/app/lib/graphql-client/config.ts#L11), it is passing in any case even if SITECORE_API_KEY env variable is not define, this is a confusing part for me

Reply:
@illiakovalenko Yeah, that is confusing... I can see now how moving this to the `disconnected` plugin makes sense. Then can we just make the `disconnected` order later than `scjssconfig` to solve the order-of-operation problem for angular (it's [2 for nextjs](https://github.com/Sitecore/jss/blob/dev/packages/create-sitecore-jss/src/templates/nextjs-styleguide/scripts/config/plugins/disconnected.ts#L13))?

Actually, not even sure the `disconnected` plugin is even necessary for angular. The only thing here is a warning message about incompatibility with GraphQL dictionary/layout with disconnected mode, but that's not even possible in angular, right??

Reply:
1 - Makes sense, I will update order. For next.js has a correct order already, as you mentioned
2 - Disconnected plugins is still needed for angular (angular-sxp), since you can use disconnected mode and potentially use GraphQL fetchWith method